src/Microsoft.PowerShell.Commands.Utility/commands/utility/join-string.cs

```diff
@@ -0,0 +1,210 @@
+
+                for (var index = 1; index < _inputObjects.Count; index++)
+                {
+                    builder.Append(Delimiter);
+                    AppendValue(GetPropertyValueString(_inputObjects[index], propertyName));
+                }
+            }
+            else if (Property is ScriptBlock sb)
+            {
+                string GetScriptBlockResultString(PSObject input)
+                {
+                    var result = sb.DoInvokeReturnAsIs(
+                        useLocalScope: true,
+                        errorHandlingBehavior: ScriptBlock.ErrorHandlingBehavior.WriteToCurrentErrorPipe,
+                        dollarUnder: input,
+                        input: AutomationNull.Value,
+                        scriptThis: AutomationNull.Value,
+                        args: Utils.EmptyArray<object>());
+                    return LanguagePrimitives.ConvertTo<string>(result);
+                }
+
+                AppendValue(GetScriptBlockResultString(_inputObjects[0]));
+
+                for (var index = 1; index < _inputObjects.Count; index++)
+                {
+                    builder.Append(Delimiter);
+                    AppendValue(GetScriptBlockResultString(_inputObjects[index]));
+                }
+            }
+            else
+            {
+                throw new ArgumentException();
+            }
+
+            builder.Append(Suffix);
+            WriteObject(builder.ToString(), false);
+        }
+    }
+
+    internal class JoinItemCompleter : IArgumentCompleter
+    {
+        public IEnumerable<CompletionResult> CompleteArgument(
+            string commandName,
+            string parameterName,
+            string wordToComplete,
+            CommandAst commandAst,
+            IDictionary fakeBoundParameters)
+        {
+            var res = new List<CompletionResult>(10);
+
+            void AddMatching(string completionText, string listText, string toolTip)
+            {
+                if (completionText.StartsWith(wordToComplete, StringComparison.OrdinalIgnoreCase))
+                {
+                    res.Add(new CompletionResult(completionText, listText, CompletionResultType.ParameterValue, toolTip));
+                }
+            }
+
+            AddMatching("', '", "Comma-Space", "', ' - Comma-Space");
+            AddMatching("';'", "Semi-Colon", "';'  - Semi-Colon ");
+            AddMatching("'; '", "Semi-Colon-Space", "'; ' - Semi-Colon-Space");
+            AddMatching($"\"{NewLineText}\"", "Newline", $"{NewLineText} - Newline");
+            AddMatching("','", "Comma", "','  - Comma");
+            AddMatching("'-'", "Dash", "'-'  - Dash");
+            AddMatching("' '", "Space", "' '  - Space");
+            return res;
+        }
+
+        public string NewLineText
+        {
+            get
+            {
+#if UNIX
+                return Platform.IsMacOS ? "`r" : "`n";
```

Comment:
It is only for very old MacOs versions. Please remove.
CoreFX doesn't use this https://source.dot.net/#System.Private.CoreLib/src/System/Environment.cs,261
For reference https://en.wikipedia.org/wiki/Newline

Reply:
Please address the comment.